mlir/include/mlir/IR/ImplicitLocOpBuilder.h

```diff
@@ -107,6 +107,68 @@ class ImplicitLocOpBuilder : public mlir::OpBuilder {
   Location curLoc;
 };
 
+/// When using the conversion framework, we need to use a reference to the old
+/// builder because of its internal state. Despite that it behaves almost the
+/// same as ImplicitLocOpBuilder.
+struct BuilderReferenceWithLoc {
+  // Constructor for analysis (no code generation, get builder disabled).
```

Comment:
What are useful things you can do with a BuilderReferenceWithLoc if there is no builder?
I'm wondering whether we should or should not make the member a proper reference instead of a pointer and thus forbid this to have a null builder.